misc/hooks/golangci-lint

```diff
@@ -14,7 +14,7 @@
 # limitations under the License.
 
 # Required version of golangci-lint
-REQUIRED_VERSION="1.62.2"
+REQUIRED_VERSION="v2.4.0"
```

Comment:
when I did the first pass, I installed manually first with go get and it was not using v2, so it was a very old version.